components/raftstore/src/store/async_io/write.rs

```diff
@@ -499,9 +514,10 @@ where
     }
 
     fn after_write_to_kv_db(&mut self, metrics: &StoreWriteMetrics) {
-        if metrics.waterfall_metrics {
-            let now = std::time::Instant::now();
-            for task in &self.tasks {
+        let now = std::time::Instant::now();
+        for task in &mut self.tasks {
```

Comment:
In most cases, there will be no callback. Better use some way to avoid looping.

Reply:
yes, I will collect these hook  to  one vector on calling `add_write_task`